contrib/babelfishpg_tsql/src/tsql_for/forjson.c

```diff
@@ -265,12 +305,270 @@ tsql_row_to_json(StringInfo state, Datum record, bool include_null_values)
 				datatype_oid = NUMERICOID;
 			}
 		}
+		
+		// Check for NULL
+		if (isnull && include_null_values)	{
+			value = palloc(sizeof(JsonbValue));
+			value->type=jbvNull;
+		}
+		else	{
+			// Extract the colummn value in the correct format
+			value = palloc(sizeof(JsonbValue));
+			jsonb_get_value(colval, isnull, value, datatype_oid);
+			value = &value->val.array.elems[0];
+		}
+
+		// Determine if the value should be inserted as a nested json object
+		parts = determine_parts(colname, &num);
+		nestedVal = value;
+
+		found = false;
+		if (num > 1)	{
+			for (int i = num - 1; i >= 0; i--)	{
+				hashKey = build_key(parts, i);
+
+				// Check if the current key exists in the hashTable
+				hashEntry = (JsonbEntry *) hash_search(jsonbHash, hashKey, HASH_FIND, &found);
+
+				// If it exists, we insert the value into the existing JsonbValue and break out of the loop
+				if (hashEntry)	{
+					// function call
+					current = hashEntry->value;
+					insert_existing_json(current, hashEntry->parent, nestedVal, hashEntry->idx, colname);
+					pfree(hashKey);
+					break;
+				}
+
+				// If it does not exist
+				hashEntry = (JsonbEntry *) hash_search(jsonbHash, (void *) hashKey, HASH_ENTER, NULL);
+				strlcpy(hashEntry->path, hashKey, NAMEDATALEN);
```

Comment:
The antlr parser enforces a 63 character limit on IDENT strings that it reads. If the path is >= 64 characters it will hashed it to be 63 characters. This is a general problem that affects multiple JSON functions, and would require modifying the way we initially parse the input. This will prevent the path from being greater than NAMEDATALEN, and I will create a jira to track fixing the 64 character bug issue